openvdb_points/tools/AttributeArray.h

```diff
@@ -922,9 +959,9 @@ template<typename ValueType_, typename Codec_>
 typename TypedAttributeArray<ValueType_, Codec_>::ValueType
 TypedAttributeArray<ValueType_, Codec_>::get(Index n) const
 {
-    if (this->isCompressed())           const_cast<TypedAttributeArray*>(this)->decompress();
-    else if (this->isOutOfCore())       this->doLoad();
-    else if (n >= this->size())         OPENVDB_THROW(IndexError, "Out-of-range access.");
+    if (this->isCompressed())                       const_cast<TypedAttributeArray*>(this)->decompress();
+    else if (this->isOutOfCore())                   this->doLoad();
+    else if (n >= this->size() * this->stride())    OPENVDB_THROW(IndexError, "Out-of-range access.");
```

Comment:
getUnsafe asserts that n is valid, but maybe the last branch here should be a separate if statement if we want the verbosity of the runtime error for compressed or out of core accesses